File: cowork-admin/src/main/java/com/shiroyk/cowork/coworkadmin/repository/doc/DocRepository.java
package com.shiroyk.cowork.coworkadmin.repository.doc;

import com.shiroyk.cowork.coworkcommon.model.doc.Doc;
import org.springframework.data.domain.Pageable;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface DocRepository extends MongoRepository<Doc, String> {
    @Query("{ 'delete': false, 'owner.id': ?0 }")
    List<Doc> findDocsByDeleteFalseAndOwner(String owner, Pageable pageable);

    @Query("{ 'owner.id': ?0 }")
    List<Doc> findDocsByOwner(String owner, Pageable pageable);
}

File: cowork-doc/service/src/main/kotlin/doc/service/repository/DocRepository.kt
package doc.service.repository

import doc.service.entity.DocEntity
import doc.service.entity.DocNode
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface DocRepository : MongoRepository<DocEntity, String>

@Repository
interface DocNodeRepository : MongoRepository<DocNode, String> {
    fun deleteAllByDid(did: String)
}